src/Bicep.Core/TypeSystem/TypeProperty.cs

```diff
@@ -5,15 +5,18 @@ namespace Bicep.Core.TypeSystem
 {
     public class TypeProperty
     {
-        public TypeProperty(string name, ITypeReference typeReference, TypePropertyFlags flags = TypePropertyFlags.None)
+        public TypeProperty(string name, ITypeReference typeReference, TypePropertyFlags flags = TypePropertyFlags.None, string? description = null)
```

Comment:
I'd like to avoid using the default parameter value here, but don't want to add noise to this PR. I may submit this as a follow-up PR.